bootstrap/kubeadm/api/v1beta1/kubeadm_types.go

```diff
@@ -429,18 +452,22 @@ type LocalEtcd struct {
 type ExternalEtcd struct {
 	// endpoints of etcd members. Required for ExternalEtcd.
 	// +kubebuilder:validation:MaxItems=50
+	// +kubebuilder:validation:items:MaxLength=256
```

Comment:
Does etcd support DNS for endpoints, or only IP addresses? If IP addresses, this could be much shorter. IPv6 + port number would be about 45 chars in the worst case

Reply:
I don't know to be honest

Reply:
Not sure if it's a question of if etcd supports it. This is set on the apiserver via etcd-servers: https://github.com/kubernetes/kubernetes/blob/85384fe273c97e07f6f7f073768d1d51452e6675/cmd/kubeadm/app/phases/controlplane/manifests.go#L198

Reply:
Ahh right, then it's possible that the etcd are DNS resolved by KAS, in which case, I'd expect scheme + DNS name + port to be specified, which might be more than 256 if the DNS name is the full 253 chars right?

Reply:
The doc mentions: 
> List of etcd servers to connect with (scheme://ip:port), comma separated.

But then in the code there are random examples like this:
> https://github.com/kubernetes/kubernetes/blob/cd5b27304f58f6b634be800ec4ec9990d28550a3/staging/src/k8s.io/apiserver/pkg/server/storage/storage_factory.go#L35

Etcd client Config
> 	// Endpoints is a list of URLs.
>	Endpoints []string `json:"endpoints"`


I would keep it as is, I just don't want to break anyone

Reply:
> Ahh right, then it's possible that the etcd are DNS resolved by KAS, in which case, I'd expect scheme + DNS name + port to be specified, which might be more than 256 if the DNS name is the full 253 chars right?

Good point. So let's bump further? What would you suggest?

Reply:
Realistically, port is at most 5 chars plus separator, and scheme is `https://` (8), so strictly we need 253 + 6 + 8 = 267, so if we want to give that a little headroom I'm ok with that, perhaps going to 300, or even 512. Though I'm pretty sure 512 is way more than anyone could validly need here, my only uncertainty really is about the scheme, could there be a longer scheme?

Reply:
Let's be safe and use 512. Not much downside to allowing that I think